ts/src/models/transaction_model.ts

```diff
@@ -44,7 +45,13 @@ export const transactionModel = {
             .leftJoinAndSelect('transaction.takerAssetFillAmounts', 'takerAssetFillAmount')
             .where('order.hash IN (:...orderHashes)', { orderHashes });
         if (opts !== undefined && opts.takerAddress !== undefined) {
-            query = query.andWhere('transaction.takerAddress = :takerAddress', { takerAddress: opts.takerAddress });
+            let queryString = 'transaction.takerAddress = :takerAddress';
+            let querySubstitutions: { takerAddress: string; txOrigin?: string } = { takerAddress: opts.takerAddress };
+            if (opts.txOrigin !== undefined) {
+                queryString += ' OR transaction.txOrigin = :txOrigin';
```

Comment:
In the case where the taker contract is whitelisted, we should _only_ query by `txOrigin`.